spring-data-microservice-transaction-sample/README.md

```diff
@@ -0,0 +1,548 @@
+  customerRepository.resume(txId);
+}
+```
+
+`join()` is called via [execTwoPcOperation()](customer-service/src/main/java/sample/customer/CustomerService.java#L99):
+
+```java
+public void payment(PaymentRequest request, StreamObserver<Empty> responseObserver) {
+  execTwoPcOperation(request.getTransactionId(), true, responseObserver, "Payment", () -> {
+```
+
+It then gets the customer information, and checks if the customer's credit total exceeds the credit limit after the payment.
+And if the check is okay, it updates the customer's credit total (the code is [here](customer-service/src/main/java/sample/customer/CustomerService.java#L100-L114)):
+
+```java
+Customer customer = getCustomer(responseObserver, request.getCustomerId());
+
+int updatedCreditTotal = customer.creditTotal + request.getAmount();
+// Check if the credit total exceeds the credit limit after payment
+if (updatedCreditTotal > customer.creditLimit) {
+  String message = String.format(
+      "Credit limit exceeded. creditTotal:%d, payment:%d", customer.creditTotal, request.getAmount());
+  responseObserver.onError(
+      Status.FAILED_PRECONDITION.withDescription(message).asRuntimeException());
+  throw new ScalarDbNonTransientException(message);
+}
+
+// Reduce credit_total for the customer
+customerRepository.update(customer.withCreditTotal(updatedCreditTotal));
+```
+
+### 3. Two-phase Commit
+
+Once Order Service receives a response that the payment succeeded, Order Service tries to commit the transaction.
+
+`executeTwoPcTransaction()` API, called on Order Service, automatically performs preparations, validations and commits of both the local Order Service and the remote Customer Serivice. These steps are executed sequentially after the above CRUD operations successfully finish. The implementations to invoke `prepare`, `validate` and `commit` gRPC endpoints of Customer Service need to be passed as parameters to the API (the code is [here](order-service/src/main/java/sample/order/OrderService.java#L134-L141)):
+
+```java
+Collections.singletonList(
+  RemotePrepareCommitPhaseOperations.createSerializable(
+    this::callPrepareEndpoint,
+    this::callValidateEndpoint,
+    this::callCommitEndpoint,
+    this::callRollbackEndpoint
+  )
+)
+```
+
+![Sequence Diagram of High Level 2PC API](images/seq-diagram-high-level-2pc-api.png)
+
+In the `prepare` endpoint of Customer Service, it resumes and prepares the transaction (the code is [here](customer-service/src/main/java/sample/customer/CustomerService.java#L122-L126)):
+
+```java
+execTwoPcOperation(request.getTransactionId(), false, responseObserver, "Payment", () -> {
+  // Prepare the transaction
+  customerRepository.prepare();
+});
+```
+
+The transaction is resumed in `execTwoPcOperation()` as shown above.
+
+
+In the `validate` endpoint of Customer Service, it resumes and validates the transaction (the code is [here](customer-service/src/main/java/sample/customer/CustomerService.java#L131-L135)):
```

Comment:
According to a [similar sample doc](https://scalardb.scalar-labs.com/docs/latest/scalardb-samples/microservice-transaction-sample/README/#error-handling), "it" seems to refer to "Customer Service". If that is correct, can we change this sentence as follows?

```suggestion
In the `validate` endpoint of Customer Service, the microservice resumes and validates the transaction (the code is [here](customer-service/src/main/java/sample/customer/CustomerService.java#L131-L135)):
```